lib/realtime/gen_counter/gen_counter.ex

```diff
@@ -162,11 +163,23 @@ defmodule Realtime.GenCounter do
     end
   end
 
+  @spec find_counter(term) :: {:ok, reference()} | {:error, :counter_not_found}
+  def find_counter(term) do
+    id = :erlang.phash2(term)
+
+    case Registry.lookup(Realtime.Registry.Unique, {__MODULE__, :counter, id}) do
+      [{pid, counter_ref}] -> {:ok, counter_ref, pid}
+      _error -> {:error, :counter_not_found}
+    end
+  end
+
   # Callbacks
 
   @impl true
   def init(args) do
-    id = Keyword.get(args, :id)
+    tenant_id = Keyword.fetch!(args, :tenant_id)
+    id = Keyword.fetch!(args, :id)
+    Logger.metadata(external_id: tenant_id, project: tenant_id)
```

Comment:
Still should not have any reference to tenant_id in here.

Should just `Logger.metadata(metadata)` where metadata is passed in with the telemetry metadata.